test/Response/SapiStreamEmitterTest.php

```diff
@@ -57,76 +61,322 @@ public function testDoesNotInjectContentLengthHeaderIfStreamSizeIsUnknown()
         }
     }
 
-    public function emitBodyProvider()
+    private function expandTestData()
     {
-        return [
-            [true,   '01234567890123456789'    , 10, 2],
-            [true,   '012345678901234567890123', 10, 3],
-            [false,  '01234567890123456789'    , 10, 2],
-            [false,  '012345678901234567890123', 10, 3],
+        $initialData = func_get_args();
+
+        $initialData = array_filter($initialData, 'is_array');
+
+        $resultData = [];
+
+        $closureExpand = function ($item, $key, $parameters) use (& $closureExpand) {
+            if (! is_array($item)) {
+                $item = [$item];
+            }
+
+            $parameters['items'] = array_merge($parameters['items'], $item);
+
+            if (count($parameters['initialData']) > 0) {
+                array_walk(array_shift($parameters['initialData']), $closureExpand, $parameters);
+            } else {
+                $parameters['resultData'][] = $parameters['items'];
+            }
+        };
+
+        array_walk(array_shift($initialData), $closureExpand, [
+            'initialData' => $initialData,
+            'items' => [],
+            'resultData' => & $resultData]);
+
+        return $resultData;
+    }
+
+    private function getStreamProphecy(& $contents, & $size, & $position, & $peakBufferLength)
```

Comment:
Removed unecessary by-ref and replaced $peakBufferLength by a callable.
I think it's now become more intuitive.